src/backtrack/ReadMaze.java

```diff
@@ -0,0 +1,75 @@
+package backtrack;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ReadMaze {
+    private Path FilePath;
+    private int [][] startPos = {{0,0}};
+    private int [][] endPos = {{0,0}};
+    private String[][] inputMaze;
+    public ReadMaze(){}
+
+    public void setFilePath(String fileName){
+        FilePath = Paths.get(fileName);
+    }
+
+    public String [][] readMaze(){
+        try(LineNumberReader lr = new LineNumberReader(Files.newBufferedReader(((FilePath))))) {
+            String line;
+            int lineCount =0;
+            while  ((line = lr.readLine()) != null) {
+                int lineNumber = lr.getLineNumber();
+                String[] currentLine = line.split(" ");
+                if (lineNumber == 1) {
```

Comment:
I have two questions here:
1. Does LineNumberReader starts with 0 or 1? I think it is 0, if yes, then you should start with 0, 1, 2
2. Why can you just initialise you double array like this:
``
int width = Integer.parseInt(currentLine[0])
int height = Integer.parseInt(currentLine[1])
 String[][] inputMaze = new String[width][height];
``

Reply:
1) From what I understand of this, it does start at line 0. However the loop is calling readLine, and from what I understand whenever a line is read from the line of text, then the lineNumber is incremented by one. So at the start of the file, we would be at line 0, but when you read this line the line number is set to 1. I could be wrong however - https://docs.oracle.com/javase/7/docs/api/java/io/LineNumberReader.html

2) I have changed the way this is initialised now-  as I think it is better the way you have shown